packages/react-network/src/hooks.ts

```diff
@@ -28,7 +29,22 @@ export function useCspDirective(
 
 export function useRequestHeader(header: string) {
   const network = useContext(NetworkContext);
-  return network ? network.getHeader(header) : undefined;
+
+  if (network) {
+    // Server: get it from context
+    // It should also store/serialize it for later client-side renders
+    return network.getHeader(header);
+  } else {
+    // Client: get it from serialized data
+    // If not present (i.e. component was not initially rendered on server) return undefined
+    try {
+      return (getSerialized('quilt-data') as {
```

Comment:
This will technically work, but using `getSerialized` here will do a bunch of DOM parsing shenanigans inline in the render loop every time the component rerenders. As a result I don't think we want to go with this approach to getting the data from serialized. Instead we likely want to use a context provider and the `useSerialized` hook.